### ISSUE TITLE
Store a new record with an unknown value that's determined by other known values

### ISSUE THREAD

**Issue by Hxnr** [Laravel-Backpack/CRUD#2056]
I am wondering if this is possible.

When inserting a new record, I have the following columns (as example):

CountryID TownID StateID ZipID
1                  1            1            2
2                  2            2            4
3                  3            3            1

( If formatting is bad for table above: https://prnt.sc/p6yqps )

When a user is storing a new record in the 'Country' Table, they know the StateID and ZipID. However, they will not know the TownID. My question, is it possible to insert/determine TownID automatically if they enter a unique StateID and ZipID? Say they enter '2' for StateID and '4' for ZipID, it would insert '2' for TownID. 

I have this issue on a much larger scale, but if it could be explained for the more simplistic example I provided that would be great.

**Comment by tabacitu** [comment]
It should be possible, yes. If you want it added without any visual interface, you can do it using (A) [Eloquent mutators](https://laravel.com/docs/6.x/eloquent-mutators) on your model or (B) using [the ```store()``` and ```update()``` methods in your EntityCrudController, before the actual saving](https://backpackforlaravel.com/docs/3.6/crud-operation-update#callbacks).

If you want an interface for it (drop-down for Town to be updated once Country is selected)  you can use the [```select2_from_ajax``` field type](https://backpackforlaravel.com/docs/3.6/crud-fields#select2-from-ajax), which has the ```dependencies``` and ```include_all_form_fields``` attributes. It basically allows you to create an AJAX response however you want it (including with an ID for a certain field). If you need the form input updated with the ID you give it with AJAX (not just cleared) you might need to create a custom input starting from ```select2_from_ajax```, that also updates the ID with the ID you've given it.

TLDR: it's possible, but if you want an interface for it, it's not a built-in feature.

Hope it helps.
Cheers!

**Comment by Hxnr** [comment]
Thank you @tabacitu, I was able to do this with Eloquent in my store method.
